seqr/views/apis/data_manager_api.py

```diff
@@ -387,14 +387,68 @@ def load_rna_seq_sample_data(request, sample_guid):
         data_by_gene = json.loads(row.split('\t\t')[1])
 
     model_cls = RNA_DATA_TYPE_CONFIGS[data_type]['model_class']
-    models = model_cls.objects.bulk_create([model_cls(sample=sample, **data) for data in data_by_gene.values()])
-    logger.info(f'create {len(models)} {model_cls.__name__}', request.user, db_update={
-        'dbEntity': model_cls.__name__, 'numEntities': len(models), 'parentEntityIds': [sample_guid], 'updateType': 'bulk_create',
-    })
+    model_cls.bulk_create(request.user, [model_cls(sample=sample, **data) for data in data_by_gene.values()], parent='sample')
 
     return create_json_response({'success': True})
 
 
+def _log_append_info(user, info, message):
+    info.append(message)
+    logger.info(message, user)
+
+
+def _load_phenotype_prioritization(file_path, user):
+    tool, data_by_project_sample_id = load_phenotype_prioritization_data_file(file_path)
+
+    info = []
+    _log_append_info(user, info, f'Parsed {tool.upper()} data for project(s): {", ".join(data_by_project_sample_id.keys())}')
+
+    all_records = []
+    to_delete = None
+    for project, records_by_sample in data_by_project_sample_id.items():
+        indivs = Individual.objects.filter(family__project__name=project, individual_id__in=records_by_sample.keys())
+        existing_indivs_by_id = {ind.individual_id: ind for ind in indivs}
+
+        missing_individuals = set(records_by_sample.keys()) - set(existing_indivs_by_id.keys())
+        if missing_individuals:
+            raise ValueError(f'Can\'t find individuals {", ".join(sorted(list(missing_individuals)))}')
+        for sample_id, records in records_by_sample.items():
+            for rec in records:
+                rec['individual'] = existing_indivs_by_id[sample_id]
+
+        exist_records = PhenotypePrioritization.objects.filter(tool=tool, individual__in=indivs)
+        to_delete = to_delete | exist_records if to_delete else exist_records
+
+        records = [rec for records in records_by_sample.values() for rec in records]
+        _log_append_info(user, info, f'Attempted loading {len(records)} records of {tool.upper()} data to project {project}')
+        all_records += records
+
+    if to_delete:
+        deleted, _ = PhenotypePrioritization.bulk_delete(user, to_delete, parent='individual')
+        _log_append_info(user, info, f'Deleted {deleted} existing {tool.upper()} records')
+
+    return all_records, info, tool
+
+
+@data_manager_required
+def load_phenotype_prioritization_data(request):
+    request_json = json.loads(request.body)
+
+    file_name = request_json['file']
+
+    logger.info(f'Loading phenotype-based prioritization data from {file_name}', request.user)
+    records, info, tool = _load_phenotype_prioritization(file_name, request.user)
```

Comment:
This helper function doesn't really help. All the logic is essentially in the helper function and its only used once, so its just a bit less readable. A better way to do a helper function would be to make the part inside the for loop the helper and hkeep the main login here